### DISCUSSION TITLE
preventing lost work if accidentally clicking outside of the modal

### DISCUSSION THREAD

**Discussion by elainelylee** [twilio-labs/paste#791] [Q&A] [answered]
Because the modal allows you to close it by clicking outside of it, there's potential for lost work if you accidentally click outside of the modal. What's the recommendation on giving a chance for someone to save their work before closing out or close without saving?

**Comment by serifluous** [comment, marked as answer]
Hey @elainelylee ! We recommend using a browser dialog (https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm), which you could trigger when someone tries to close the modal (regardless of whether they click outside of it, press the Esc key, etc).

**Reply by TheSisb** [reply]
Another option rather than displaying a confirm dialog could be to save the current form field values to localstorage or sessionstorage in the `onDismiss` function. Then you can restore the values when the modal is reopened. How to do that is up to the implementing product team, based on your use-case and needs.

**Comment by elainelylee** [comment]
Great! Alternatively, I started exploring having certain entries
automatically saved and breaking apart the steps in different modal
windows.

Elaine Lee
Principal Product Designer - Twilio AI
[image: twilio] <https://www.twilio.com/?utm_source=email_signature>
EMAIL ellee@twilio.com

**Reply by serifluous** [reply]
Very cool! Not sure if you're planning on doing something like a multi-step modal, but @andipants12 on V&V is prototyping one with the Paste modal component in Discussion #790. If you follow along there, your team may be able to use the same thing.

**Reply by pguptaDS** [reply]
@serifluous I am curious to understand the reasoning behind the default behavior of closing the modal if the user clicks outside of it.

**Reply by serifluous** [reply]
Hey @pguptaDS ! From looking at the original [design brief](https://docs.google.com/document/d/1GqUaD94MFwICZN8vcQkGZXLv4BQ-9s6IgD75y-0rbcA/edit?usp=sharing) (cmd+f for references of "click"), it seems like it may have been because legacy console and SendGrid modals had the same interaction. @kylegilbreath might have more context since he contributed the component design.

**Reply by pguptaDS** [reply]
Just circling back on this. Would love to understand the reasoning behind this (cc @kylegilbreath) and curious if that reasoning stands true today as well.
